OneOf.Serialization.Tests/OneOfJsonConverterTests.cs

```diff
@@ -44,13 +83,61 @@ public void Given_Json_With_OneOf_Case_When_Deserializing_Then_Produces_Correct_
         {
             var json = File.ReadAllText("./working-engine.json");
             var engine = JsonConvert.DeserializeObject<Engine>(json);
-            var engineStatus = (engine.Value as WorkingEngine).Status;
             engine.Should().NotBeNull();
             engine.Value.Should().BeOfType<WorkingEngine>();
+            var engineStatus = (engine.Value as WorkingEngine).Status;
             engineStatus.Should().NotBeNull();
             engineStatus.Value.Should().BeOfType<Status.Idle>();
         }
 
+        [Fact]
+        public void Given_Json_Array_Of_OneOf_Cases_When_Deserializing_Then_Produces_Correct_Enumerable_Instance_Of_Cases()
+        {
+            var json = File.ReadAllText("./working-engine-array.json");
+            var engineEnumerable = JsonConvert.DeserializeObject<IEnumerable<Engine>>(json);
+            engineEnumerable.Should().NotBeNull();
+
+            var engineList = engineEnumerable.ToList();
+
+            var engine0 = engineList[0];
+            engine0.Should().NotBeNull();
+            engine0.Value.Should().BeOfType<WorkingEngine>();
+            var engine0Status = (engine0.Value as WorkingEngine).Status;
+            engine0Status.Should().NotBeNull();
+            engine0Status.Value.Should().BeOfType<Status.Idle>();
```

Comment:
Does it make sense to create a tiny extension method hiding the details and expressing something like:
```cs
workingEngine0.ShouldBeIdle();
```

:thinking:

Reply:
That's maybe up to you. I've only copied that code. But for symmetry, a similar method `workingEngine1.ShouldBeStarted(readiness)` should be created.

Making similar extension methods would be a bit more difficult for my next pull request though, which contains significantly more code.